sdk/servicebus/service-bus/samples/typescript/src/receiveMessagesStreaming.ts

```diff
@@ -30,26 +36,49 @@ export async function main() {
   // - See session.ts for how to receive using sessions.
   const receiver = sbClient.createReceiver(queueName);
 
-  const processMessage = async (brokeredMessage) => {
-    console.log(`Received message: ${brokeredMessage.body}`);
-    await brokeredMessage.complete();
-  };
-  const processError = async (args) => {
-    console.log(`Error from error source ${args.errorSource} occurred: `, args.error);
-  };
-
   try {
-    receiver.subscribe(
+    const subscription = receiver.subscribe(
       {
-        processMessage,
-        processError
+        processMessage: async (brokeredMessage: ServiceBusReceivedMessage) => {
+          console.log(`Received message: ${brokeredMessage.body}`);
+          await receiver.completeMessage(brokeredMessage);
+        },
+        processError: async (args: ProcessErrorArgs) => {
+          console.log(`Error from source ${args.errorSource} occurred: `, args.error);
+
+          // the handler will not stop without explicit intervention from you.
+          if (isMessagingError(args.error)) {
```

Comment:
Just putting it out there.. I was getting the args.error to be `MessagingError<false>` by default and I'm left with no IntelliSense.
If we are going this way, I'd like to get that resolved first.

Reply:
I'm removing the fancy type-ness for now but we'll see what we can do here. 

I think we can side-step this altogether by doing what I had originally (just having a specific Error class for ServiceBus).